README.md

```diff
@@ -1 +1,314 @@
+        (web100_log_entry.snap.SndLimTimeRwin +
+         web100_log_entry.snap.SndLimTimeCwnd +
+         web100_log_entry.snap.SndLimTimeSnd)) AS download_mbps,
+  (web100_log_entry.snap.SumRTT /
+     web100_log_entry.snap.CountRTT) AS avg_rtt,
+  web100_log_entry.snap.MinRTT AS min_rtt,
+  (web100_log_entry.snap.SegsRetrans /
+     web100_log_entry.snap.DataSegsOut) AS packet_retransmit_rate
+FROM
+  [plx.google:m_lab.ndt.all]
+WHERE
+  web100_log_entry.snap.SndLimTimeSnd IS NOT NULL
+  AND web100_log_entry.snap.SndLimTimeCwnd IS NOT NULL
+  AND web100_log_entry.snap.SndLimTimeRwin IS NOT NULL
+  AND project = 0
+  AND web100_log_entry.is_last_entry = True
+  AND connection_spec.data_direction = 1
+  AND web100_log_entry.snap.CongSignals > 0
+  AND web100_log_entry.snap.HCThruOctetsAcked >= 8192
+  AND (web100_log_entry.snap.State == 1
+    OR (web100_log_entry.snap.State >= 5
+        AND web100_log_entry.snap.State <= 11))
+  AND (web100_log_entry.snap.SndLimTimeRwin +
+       web100_log_entry.snap.SndLimTimeCwnd +
+       web100_log_entry.snap.SndLimTimeSnd) >= 9000000
+  AND (web100_log_entry.snap.SndLimTimeRwin +
+       web100_log_entry.snap.SndLimTimeCwnd +
+       web100_log_entry.snap.SndLimTimeSnd) < 3600000000
+  AND web100_log_entry.snap.CountRTT > 10
+```
+
+## Using computers
+
+We specify a signal by a series of transforms, roll-ups, and drill downs. So
+the space we are searching for interesting signals is the set of all possible
+transforms, roll-ups, and drill-downs applied to each dimension in our data
+set. The size of this search space grows exponentially with the number of
+dimensions a measurement has ("the curse of dimensionality") and
+multiplicatively with the number of values in each dimension.  Even with
+computational help, this is too large a space to use brute force. Instead, we
+will specify roll-ups and drill-downs of interest and then use the derived time
+series data.
+
+To cover the project's remit of general search, we will train an ML
+model on our data and then run any new data through the model to see if the new
+results are mispredicted (and therefore "surprising", which can be thought of
+as "potentially interesting").
+
+# Design
+
+We use this insight to build a system which goes through our database
+looking for data streams of interest that violate at least one of
+these principles.  Because of our focus on end-to-end user experience,
+we restrict ourselves to searching for signals in metrics which affect
+the end-user experience: Download speed, upload speed, and RTT.
+
+Expanding on the initial super-generic statement, we come up with the
+following assertions about the properties a healthy and open Internet
+should have:
+
+1. Performance for a given ISP or ISP pair should not vary diurnally.
+   Diurnal performance variation was discovered to be a leading indicator
+   of network congestion problems.
+
+2. Performance should not get worse over time. If we ever find
```

Comment:
I don't *think* it is? Signal-searcher's intended audience is people looking for problems, so "performance improved over here" isn't really news. Even more, in a healthy Internet, performance should be steadily improving over time. Massive improvements might be interesting, but I'd like at least version 1 to focus on finding problems.